paddle/fluid/framework/attribute.h

```diff
@@ -165,6 +165,35 @@ struct ExtractAttribute<float> {
   const std::string& attr_name_;
 };
 
+template <>
+struct ExtractAttribute<std::vector<double>> {
+  explicit ExtractAttribute(const std::string& attr_name)
+      : attr_name_(attr_name) {}
+
+  std::vector<double>* operator()(Attribute& attr) const {
+    if (attr.type() == typeid(std::vector<int>)) {  // NOLINT
+      std::vector<int> val = BOOST_GET_CONST(std::vector<int>, attr);
+      std::vector<double> vec(val.begin(), val.end());
```

Comment:
This code is inefficient because line 176 will construct `vec` by iterating the vector and then `attr = vec` will do an iterating copy.

Can you convert `val` to `attr` inplace so that it just iterates once?

Same as line 179 - line 181

Reply:
Thanks.